kmip/tests/unit/core/messages/test_messages.py

```diff
@@ -1775,6 +1963,175 @@ def test_locate_response_write(self):
+                 b'\x32\x9f'
+                 b'\x6a\xc4\xed\x97\x6e\xac\x99\xe5\x21\x53\xc4\x19\x28\xf2'
+                 b'\x2a\x5b'
+                 b'\xef\x79\xa4\xbe\x05\x3b\x31\x49\x19\xe0\x75\x23\xb9\xbe'
+                 b'\xc8\x23'
+                 b'\x35\x60\x7e\x49\xba\xa9\x7e\xe0\x9e\x6b\x3d\x55\xf4\x51'
+                 b'\xff\x7c')
+            self.assertIsInstance(mac_data, objects.MACData,
+                                  self.msg.format('secret', 'type',
+                                                  objects.MACData,
+                                                  type(mac_data)))
+            self.assertEqual(value, mac_data.value,
+                             self.msg.format('mac data', 'value',
+                                             binascii.hexlify(mac_data.value),
+                                             binascii.hexlify(value)))
+
+    def test_mac_response_write(self):
+        prot_ver = contents.ProtocolVersion.create(1, 2)
+
+        # Fri Apr 27 10:12:23 CEST 2012
+        time_stamp = contents.TimeStamp(0x588a3f23)
+
+        batch_count = contents.BatchCount(1)
+        response_header = messages.ResponseHeader(protocol_version=prot_ver,
+                                                  time_stamp=time_stamp,
+                                                  batch_count=batch_count)
+        operation = contents.Operation(enums.Operation.MAC)
+        result_status = contents.ResultStatus(enums.ResultStatus.SUCCESS)
+
+        uuid = '1'
+        uniq_id = attr.UniqueIdentifier(uuid)
+
+        value = \
+            (b'\x99\x8b\x55\x59\x90\x9b\x85\x87\x5b\x90\x63\x13\x12\xbb'
+             b'\x32\x9f'
+             b'\x6a\xc4\xed\x97\x6e\xac\x99\xe5\x21\x53\xc4\x19\x28\xf2'
+             b'\x2a\x5b'
+             b'\xef\x79\xa4\xbe\x05\x3b\x31\x49\x19\xe0\x75\x23\xb9\xbe'
+             b'\xc8\x23'
+             b'\x35\x60\x7e\x49\xba\xa9\x7e\xe0\x9e\x6b\x3d\x55\xf4\x51'
+             b'\xff\x7c')
+        mac_data = objects.MACData(value)
+
+        resp_pl = mac.MACResponsePayload(unique_identifier=uniq_id,
+                                         mac_data=mac_data)
+        batch_item = messages.ResponseBatchItem(operation=operation,
+                                                result_status=result_status,
+                                                response_payload=resp_pl)
+        rm = messages.ResponseMessage(response_header=response_header,
+                                      batch_items=[batch_item])
+        rm.write(self.stream)
+
+        result = self.stream.read()
+        len_exp = len(self.mac)
+        len_rcv = len(result)
+        self.assertEqual(len_exp, len_rcv,
+                         self.msg.format('get response message', 'write',
+                                         len_exp, len_rcv))
+
+        msg = "Bad response message write: encoding mismatch"
+        self.assertEqual(self.mac, result, msg)
+
```

Comment:
I'm fine with keeping these additions to `test_messages.py`. However we do need a dedicated `payloads/test_mac.py` just for the MAC payloads. See [GetAttributesList](https://github.com/OpenKMIP/PyKMIP/blob/master/kmip/tests/unit/core/messages/payloads/test_get_attribute_list.py) and [GetAttributes](https://github.com/OpenKMIP/PyKMIP/blob/master/kmip/tests/unit/core/messages/payloads/test_get_attributes.py) for examples.